SnipperAPI/Controllers/SnippetsControllers.cs

```diff
@@ -26,10 +32,40 @@ public ActionResult<Snippet> CreateSnippet([FromBody] SnippetRequest snippetRequ
 
             // Returning the created snippet with Created status - use abstraction / restfulAPI
             return CreatedAtAction(nameof(GetSnippet), new { id = newSnippet.Id }, newSnippet);
+        }*/
+
+
+        [HttpPost("snippets")]
+        [Authorize] // Requires authentication
+        public ActionResult<Snippet> CreateSnippet([FromBody] SnippetRequest snippetRequest)
+        {
+            if (ModelState.IsValid)
+            {
+                var currentUser = _userManager.GetUserAsync(User).Result;
```

Comment:
Is something potentially missing that may cause an null exception when this endpoint is called?

Have a look at your users controller as theres a difference between that one and this one tat would likely cause this issue the hint is this line. Let me know if you need any more info want to be vague though to allow for a bit of thinking around the issue